src/main/java/core/basesyntax/BlockingQueue.java

```diff
@@ -12,16 +12,32 @@ public BlockingQueue(int capacity) {
     }
 
     public synchronized void put(T element) throws InterruptedException {
-        // write your code here
+        while (queue.size() >= capacity) {
+            try {
+                wait();
+            } catch (InterruptedException e) {
+                throw new RuntimeException("Element couldn't be added to the collection "
+                        + "according to interrupted thread", e);
+            }
```

Comment:
we don't need to add try catch block here, we have `throws InterruptedException` in the signature of the method